crates/biome_js_analyze/src/lint/nursery/no_secrets.rs

```diff
@@ -0,0 +1,129 @@
+    r#"[gG][iI][tT][hH][uU][bB].*['\"][0-9a-zA-Z]{35,40}['\"]"#,  // GitHub
+    r#""client_secret":"[a-zA-Z0-9-_]{24}""#,  // Google OAuth
+    r"AKIA[0-9A-Z]{16}",  // AWS API Key
+    r"[hH][eE][rR][oO][kK][uU].*[0-9A-F]{8}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{12}",  // Heroku API Key
+    r#"[sS][eE][cC][rR][eE][tT].*['\"][0-9a-zA-Z]{32,45}['\"]"#,  // Generic Secret
+    r#"[aA][pP][iI][_]?[kK][eE][yY].*['\"][0-9a-zA-Z]{32,45}['\"]"#,  // Generic API Key
+    r"https://hooks\.slack\.com/services/T[a-zA-Z0-9_]{8}/B[a-zA-Z0-9_]{8}/[a-zA-Z0-9_]{24}",  // Slack Webhook
+    r#""type": "service_account""#,  // Google (GCP) Service-account
+    r"SK[a-z0-9]{32}",  // Twilio API Key
+    r#"[a-zA-Z]{3,10}://[^/\s:@]{3,20}:[^/\s:@]{3,20}@.{1,100}['"\s]"#,  // Password in URL
+];
+
+// TODO: Try to get this to work in JavaScript comments as well
+declare_lint_rule! {
+    /// Disallow usage of sensitive data such as API keys and tokens.
+    ///
+    /// This rule checks for high-entropy strings and matches common patterns
+    /// for secrets, such as AWS keys, Slack tokens, and private keys.
+    ///
+    /// Inspired by the ESLint [no-secrets/no-secrets](https://github.com/nickdeis/eslint-plugin-no-secrets) rule.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// var secret = "AKIA1234567890EXAMPLE";
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// var nonSecret = "hello world";
+    /// ```
+    pub NoSecrets {
+        version: "next",
+        name: "noSecrets",
+        language: "js",
+        recommended: false,
+        sources: &[RuleSource::Eslint("no-secrets/no-secrets")],
+        source_kind: RuleSourceKind::Inspired,
+    }
+}
+
+impl Rule for NoSecrets {
+    type Query = Ast<JsIdentifierBinding>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+
+        if let Some(string_literal) = node.syntax().first_token() {
+            let value = string_literal.text().to_string();
+
+            for pattern in SENSITIVE_PATTERNS {
+                let re = Regex::new(pattern).unwrap();
+                if re.is_match(&value) {
+                    return Some(());
+                }
+            }
+
+            if is_high_entropy(&value) {
+                return Some(());
+            }
+        }
+
+        None
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _state: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                node.range(),
+                markup! { "Potential secret found." }
+            ).note(markup! { "This looks like a sensitive value such as an API key or token." }) // TODO: Give a more detailed response on the *type* of API Key/token (based on the SENSITIVE PATTERNS)
```

Comment:
You could define your SENSITIVE_PATTERNS with a tuple `(regex, "Additional comment")` And return comment in diagnostics section. Alternatively your could use enums.

Reply:
Had considered this. I thought I'd keep it simple for now, but seems like this provides a lot of value.

Reply:
Done. Please check the latest code.